### DISCUSSION TITLE
性能測試項目是缺少還是尚未完成

### DISCUSSION THREAD

**Discussion by bookbug666** [LmeSzinc/AzurLaneAutoScript#2097] [Q&A] [answered]
看到模擬器截圖方案多了DroidCast跟Scrcpy，還沒完成就算了
想看說能不能測試或更換新的截圖方案

**Comment by LmeSzinc** [comment, marked as answer]
DroidCast, scrcpy, Hermit 都是针对特定设备添加的截图控制方式，不需要添加到性能测试里。
- Hermit 是针对安卓真机里的 vmos 虚拟机添加的，但是真机运行已经被放弃了，这个控制方式也没什么用处了。vmos 虚拟机内无法运行 minitouch，用 Hermit 贪点速度，点击是 90ms 比 minitouch 慢，滑动和拖拽稀烂需要回退到 ADB，在 vmos 外完全没优势。
- DroidCast 截图是针对 MuMu X 模拟器添加的，mumu x 下其他截图方案都是黑屏但是 DroidCast 可以，不过 mumu x 更新了几个版本之后 DroidCast 也黑了，所以也没用了。DroidCast 需要在安卓上将截图编码为 png 图片，所以速度和 ADB 差不多慢。
- scrcpy 截图也是针对 MuMu X 模拟器添加的，其他模拟器也可以用。它是传输的是视频流，目前设置是固定 6fps，延迟自然也就是固定的 167ms，没有测试的意义。视频流都是压缩过的，所以现在还有很多潜在的 bug。
